### ISSUE TITLE
API endpoint to analyse remote packages

### ISSUE THREAD

**Issue by lachlancollins** [bluwy/publint#89]
For the sveltesociety.dev website, I've taken the worker from this repo to create [this script](https://github.com/svelte-society/sveltesociety.dev/blob/main/scripts/updatePublint.js) and generate [this data](https://github.com/svelte-society/sveltesociety.dev/blob/main/src/lib/data/publint.json) on if svelte packages pass publint. However, this is pretty similar to what I imagine happens to generate https://publint.dev/analysis.json.

It would be nice if we didn't have to re-implement the VFS/unzipping/etc logic from your own repo. Would it be possible to share an endpoint to check if packages pass publint? Thanks!

**Comment by bluwy** [comment]
That's a good idea. The main concern I had was server costs as this means downloading arbitrary packages on the server-side, which instead I opt-ed to run publint on the client-side to avoid it. And I think the scope will also increase for implementing caching to save on the cost too.

Maybe tackling the issue in a different way, we could expose a package that simplifies downloading npm packages instead? There is https://www.npmjs.com/package/pacote but I didn't use it because it's bulky and I like re-implementing things 😄

And also `publint` could re-export `createTarballVfs()` from `publint/utils` to help with this.

**Comment by lachlancollins** [comment]
> That's a good idea. The main concern I had was server costs as this means downloading arbitrary packages on the server-side, which instead I opt-ed to run publint on the client-side to avoid it. And I think the scope will also increase for implementing caching to save on the cost too.
> 
> Maybe tackling the issue in a different way, we could expose a package that simplifies downloading npm packages instead? There is [npmjs.com/package/pacote](https://www.npmjs.com/package/pacote) but I didn't use it because it's bulky and I like re-implementing things 😄
> 
> And also `publint` could re-export `createTarballVfs()` from `publint/utils` to help with this.

Yeah I can imagine it would get expensive easily. Some sort of exported util would be amazing!

**Comment by bluwy** [comment]
This should be resolved in https://github.com/bluwy/publint/pull/122 and released in the next breaking minor soon. Instead of what you have [here](https://github.com/svelte-society/sveltesociety.dev/blob/265112639154495790329e5f2b65d3daa5020207/scripts/updatePublint.js#L56-L80) now, you can rewrite is as:

```js
async function processPackage(pkg) {
	const tarballUrl = getNpmTarballUrl(pkg.npm, pkg.version);
	let resultBuffer;
	try {
		const result = await fetch(tarballUrl);
		resultBuffer = await result.arrayBuffer();
	} catch (e) {
		postMessage({ type: 'error', data: 'Package not found' });
		console.error(e);
	}

	const { messages } = await publint({ pack: { tarball: resultBuffer }, level: 'warning' });
	return { [pkg.npm]: { valid: messages.length === 0 } };
}
```
